packages/relayer-lib/src/submitter/eth_eureka.rs

```diff
@@ -0,0 +1,283 @@
+//! The `ChainSubmitter` submits txs to [`EthEureka`] based on events from [`CosmosSdk`].
+
+use std::{env, str::FromStr};
+
+use alloy::{
+    primitives::Address,
+    providers::Provider,
+    sol_types::{SolCall, SolValue},
+    transports::Transport,
+};
+use anyhow::Result;
+use futures::future;
+use ibc_core_host_types::identifiers::ChainId;
+use ibc_eureka_solidity_types::{
+    ics02::client::clientInstance,
+    ics26::{
+        router::{
+            ackPacketCall, multicallCall, recvPacketCall, routerCalls, routerInstance,
+            timeoutPacketCall,
+        },
+        IICS02ClientMsgs::Height,
+        IICS26RouterMsgs::{MsgAckPacket, MsgRecvPacket, MsgTimeoutPacket},
+    },
+};
+// Re-export the `SupportedProofType` enum.
+pub use sp1_ics07_tendermint_prover::prover::SupportedProofType;
+use sp1_ics07_tendermint_prover::{
+    programs::UpdateClientAndMembershipProgram, prover::SP1ICS07TendermintProver,
+};
+
+use sp1_ics07_tendermint_solidity::{
+    sp1_ics07_tendermint,
+    IICS07TendermintMsgs::ClientState,
+    IMembershipMsgs::{MembershipProof, SP1MembershipAndUpdateClientProof},
+    ISP1Msgs::SP1Proof,
+};
+use sp1_ics07_tendermint_utils::{
+    light_block::LightBlockExt, merkle::convert_tm_to_ics_merkle_proof, rpc::TendermintRpcExt,
+};
+use sp1_sdk::HashableKey;
+use tendermint_rpc::{Client, HttpClient};
+
+use crate::{
+    chain::{CosmosSdk, EthEureka},
+    events::EurekaEvent,
+};
+
+use super::r#trait::ChainSubmitterService;
+
+/// The `ChainSubmitter` submits txs to [`EthEureka`] based on events from [`CosmosSdk`].
+#[allow(dead_code)]
+pub struct ChainSubmitter<T: Transport + Clone, P: Provider<T> + Clone> {
```

Comment:
I thought the goal of this relayer was serve an API, not to submit?
Not sure if this scope has been agreed upon

Reply:
Yes, the naming is the issue as it is not a submitter anymore. I don't know what a better name would be? Like `TxFactory`? But I can see us adding submission as a feature in the future as well. Do you think it is enough to just fix rust docs?

Reply:
Aaah! That makes sense, then. I will take another round on this file. 

If we add submission in the future, we can deal with naming issues then. For now, I think it would be better to call it something like TransactionBuilder.